test/powershell/Modules/Microsoft.PowerShell.Core/Import-Module.Tests.ps1

```diff
@@ -124,3 +124,56 @@ namespace ModuleCmdlets
     }
  }
 
+Describe "Import-Module should be case insensitive on Unix" -Tags 'CI' {
+    BeforeAll {
+        $defaultParamValues = $PSDefaultParameterValues.Clone()
+        $defaultPSModuleAutoloadingPreference = $PSModuleAutoloadingPreference
+        $originalPSModulePath = $env:PSModulePath.Clone()
+        $modulesPath = "$TestDrive\Modules"
+        $env:PSModulePath += [System.IO.Path]::PathSeparator + $modulesPath
+        $PSModuleAutoloadingPreference = "none"
+        $PSDefaultParameterValues["it:skip"] = !$IsLinux
+    }
+    AfterAll {
+        $global:PSDefaultParameterValues = $defaultParamValues
+        $global:PSModuleAutoloadingPreference = $defaultPSModuleAutoloadingPreference
+        $env:PSModulePath = $originalPSModulePath
+    }
+
+    It "Import-Module can import a module using different casing using '<modulePath>' and manifest:<manifest>" -TestCases @(
+        @{modulePath="TESTMODULE/1.1"; manifest=$true},
+        @{modulePath="TESTMODULE/1.1"; manifest=$false},
+        @{modulePath="TESTMODULE"    ; manifest=$true},
+        @{modulePath="TESTMODULE"    ; manifest=$false}
+        ) {
+        param ($modulePath, $manifest)
+        New-Item -ItemType Directory -Path "$modulesPath/$modulePath" -Force
+        if ($manifest) {
+            New-ModuleManifest -Path "$modulesPath/$modulePath/TESTMODULE.psd1" -RootModule "TESTMODULE.psm1" -ModuleVersion 1.1
+        }
+        Set-Content -Path "$modulesPath/$modulePath/TESTMODULE.psm1" -Value "function mytest { 'hello' }"
+        Import-Module testMODULE
```

Comment:
Maybe `-Force` to ensure you are not loading a cached version.

Reply:
will add

Reply:
actually, as part of this test, I'm removing the module with a different casing, so I don't want to -force loading

Reply:
Ok